torch/ao/quantization/fx/convert.py

```diff
@@ -107,51 +110,103 @@ def _replace_observer_with_quantize_dequantize_node(
-                    # for qparams that are not scale/zero_point (like axis, dtype) we store them as literals in the graph.
-                    quantize_op_inputs.append(value)
+        return
 
-            quantized_node = graph.create_node(node_type, quantize_op, tuple(quantize_op_inputs), {})
+    # otherwise, we can convert the observer module call to quantize/dequantize node
+    # 1. extract the information from activation_post_process module for generating
+    # the quantize and dequantize operator
+    dtype = activation_post_process.dtype  # type: ignore[attr-defined]
+    compute_dtype = None
+    if hasattr(activation_post_process, "compute_dtype"):
+        compute_dtype = activation_post_process.compute_dtype  # type: ignore[attr-defined]
+    quantize_op : Optional[Union[Callable, str]] = None
+    if dtype in [torch.quint8, torch.qint8, torch.qint32] and \
+            not hasattr(activation_post_process, 'compute_dtype'):
+        node_type = "call_function"
+        scale, zero_point = activation_post_process.calculate_qparams()  # type: ignore[attr-defined]
+        if is_per_channel(activation_post_process.qscheme):  # type: ignore[attr-defined]
+            ch_axis = int(activation_post_process.ch_axis)  # type: ignore[attr-defined]
+            qparams = {"_scale_": scale, "_zero_point_": zero_point, "_axis_": ch_axis, "_dtype_": dtype}
             if is_decomposed:
-                # use the same qparams from quantize op
-                dq_inputs = [quantized_node] + quantize_op_inputs[1:]
-                dequantized_node = graph.call_function(
-                    torch.ops.quantized_decomposed.dequantize_per_tensor,
-                    tuple(dq_inputs),
-                    {}
-                )
+                raise NotImplementedError("decomposed quantize_per_channel op not implemented yet")
             else:
-                dequantized_node = graph.call_method("dequantize", args=(quantized_node,))
-            node.replace_all_uses_with(dequantized_node)
-            graph.erase_node(node)
+                quantize_op = torch.quantize_per_channel
+        else:
+            scale = float(scale)
+            zero_point = int(zero_point)
+            if is_decomposed:
+                quant_min = activation_post_process.quant_min  # type: ignore[attr-defined]
+                quant_max = activation_post_process.quant_max  # type: ignore[attr-defined]
+                dtype = to_underlying_dtype(dtype)
+                qparams = {
+                    "_scale_": scale,
+                    "_zero_point_": zero_point,
+                    "_quant_min": quant_min,
+                    "_quant_max": quant_max,
+                    "_dtype_": dtype
+                }
+                quantize_op = torch.ops.quantized_decomposed.quantize_per_tensor
+            else:
+                qparams = {"_scale_": scale, "_zero_point_": zero_point, "_dtype_": dtype}
+                quantize_op = torch.quantize_per_tensor
```

Comment:
Is this code taken from `get_quantize_node_info`?

Reply:
Yes, we'll remove `get_quantize_node_info` in the other PR in this stack